src/NUnitFramework/framework/Constraints/Comparers/StringsComparer.cs

```diff
@@ -14,12 +14,10 @@ internal static class StringsComparer
             if (!(x is string xString) || !(y is string yString))
                 return null;
 
-            bool caseInsensitive = equalityComparer.IgnoreCase;
+            if (equalityComparer.IgnoreCase)
+                return xString.Equals(yString, StringComparison.CurrentCultureIgnoreCase);
```

Comment:
I think it would be nicer to do something like:
```csharp
StringComparison stringComparison = equalityComparer.IgnoreCase ? StringComparison.CurrentCultureIgnoreCase : StringComparison.CurrentCulture;
return xString.Equals(yString, stringComparison)
```
Or in one go:
```csharp
return xString.Equals(yString, equalityComparer.IgnoreCase ? StringComparison.CurrentCultureIgnoreCase : StringComparison.CurrentCulture)
```

Same for all other cases. It is less code and show the same path, just another comparison flag.

Reply:
Thanks @manfred-brands ! I've pushed some updates for this. One note: I had to use Ordinal instead of CurrentCulture for the case-sensitive path to match existing behaviour

Reply:
The tricky part is that the defaults are inconsistent. Indeed .Equals uses Ordinal, but EndsWith uses CurrentCulture. Hence there is an Analyzer that suggests to always specify a Comparison.

Reply:
Ordinal is the preferred if we can use it.

Reply:
Right, I'd forgotten about the different default. Thanks for noticing! I'll do another pass next chance I get. I think there's a case to be made for having all comparisons consistent, but I think that's a different convo for a different issue